DynamicTablesPkg/Include/ArchCommonNameSpaceObjects.h

```diff
@@ -722,6 +722,11 @@ typedef struct CmArchCommonTpm2InterfaceInfo {
 
   /** Log Area Start Address */
   UINT64    Lasa;
+
+  /** Tpm Device Size.
+   *  This information is used for Tpm2 Device Table using DynamicTableGenerator.
+   */
+  UINT64    TpmDeviceSize;

```

Comment:
I think it might be better to create a separate CmObj for TPM devices. This might be arguable from a platform PoV, but if there are multiple TPM devices, their size might differ (and their address also surely).

A TPM device object with:
- a base address (populated by `FixedPcdGet64 (PcdTpmBaseAddress)` in the configuration manager)
- an address length

should fit right ?